src/ophyd_async/core/detector.py

```diff
@@ -283,20 +295,29 @@ async def _prepare(self, value: T) -> None:
         """
         assert type(value) is TriggerInfo
         self._trigger_info = value
-        self._initial_frame = await self.writer.get_indices_written()
-        self._last_frame = self._initial_frame + self._trigger_info.number
-
-        required = self.controller.get_deadtime(self._trigger_info.livetime)
-        assert required <= self._trigger_info.deadtime, (
-            f"Detector {self.controller} needs at least {required}s deadtime, "
-            f"but trigger logic provides only {self._trigger_info.deadtime}s"
-        )
-        self._arm_status = await self.controller.arm(
-            num=self._trigger_info.number,
-            trigger=self._trigger_info.trigger,
-            exposure=self._trigger_info.livetime,
-        )
-        self._fly_start = time.monotonic()
+        if (
+            value.trigger is DetectorTrigger.internal
+            and value.deadtime is None
+            and value.livetime is None
+        ):
+            ...
+        else:
+            self._initial_frame = await self.writer.get_indices_written()
+            self._last_frame = self._initial_frame + self._trigger_info.number
+
+            if self._trigger_info.deadtime is not None:
+                required = self.controller.get_deadtime(self._trigger_info.livetime)
+                assert required <= self._trigger_info.deadtime, (
+                    f"Detector {self.controller} needs at least {required}s deadtime, "
+                    f"but trigger logic provides only {self._trigger_info.deadtime}s"
+                )
+                self._arm_status = await self.controller.arm(
+                    num=self._trigger_info.number,
+                    trigger=self._trigger_info.trigger,
+                    exposure=self._trigger_info.livetime,
+                )
```

Comment:
Logic should be:
- if trigger_info.trigger != DetectorTrigger then assert triggerinfo.deadtime
- if triggerinfo.deadtime then check it is >= self.controller.get_deadtime
- then *always* call self.controller.arm

The underlying controllers should cope with being told to arm with livetime/deadtime of None, and leave the exposure time of the detector where it was